pkg/sensors/tracing/generickprobe.go

```diff
@@ -255,6 +259,35 @@ func createMultiKprobeSensor(sensorPath string, multiIDs, multiRetIDs []idtable.
 	return progs, maps
 }
 
+// addSyscallPrefix detects if the symbol is already prefixed with arch specific
+// prefix in the form of "__x64_" or "__arm64_", and if not, adds it based on
+// provided arch.
+//
+// Note that cilium/ebpf, retries to attach to an arch specific version of the
+// symbol after a failure thus making possible to attach to "sys_listen" which
+// will fail and retry with the corrected "__arm64_sys_listen" if you are
+// running on arm64. See https://github.com/cilium/ebpf/pull/304.  However, this
+// causes Tetragon users to believe that two symbols exist (following the same
+// example) "__arm64_sys_listen" and "sys_listen", which results in differents
+// events.  Because we actually know if a call is a syscall, we can pre-format
+// the symbol directly here to allow the user to specify the "sys_listen"
+// version while only registering the real symbol "__arm64_sys_listen".
+func addSyscallPrefix(symbol string, arch string) (string, error) {
+	for _, p := range supportedArchPrefix {
+		if strings.HasPrefix(symbol, p) {
+			// check that the prefix found is the correct one
+			if p != supportedArchPrefix[arch] {
+				return "", fmt.Errorf("expecting %s and got %s", supportedArchPrefix[arch], p)
+			}
```

Comment:
nit: we can get the arch for p from the iterator and check it against `arch`

```
for p_arch, p := range supportedArchPrefix {
     if  strings.HasPrefix(symbol, p) {
         if arch != p_arch {
                    // return error
         }
}
```

Reply:
(just a suggestion, feel free to ignore)

Reply:
I see but it would not be the same code I think. `supportedArchPrefix[arch]` is fixed and is the arch of the machine passed by the second function argument. Maybe it's not super clear but on the check:
```go
if p != supportedArchPrefix[arch] {
    return "", fmt.Errorf("expecting %s and got %s", supportedArchPrefix[arch], p)
}
```
I check for the error case when you have for example a x64 prefix on and your static running arch is arm64.

Reply:
yes good idea it would simplify this if statement that was a bit confusing (that's why I added a comment) :)